src/hint_processor/builtin_hint_processor/hint_utils.rs

```diff
@@ -116,7 +116,7 @@ pub fn get_reference_from_var_name<'a>(
 ) -> Result<&'a HintReference, HintError> {
     ids_data
         .get(var_name)
-        .ok_or(HintError::UnknownIdentifier(var_name.to_string()))
+        .ok_or(HintError::UnknownIdentifier(Box::new(var_name.to_string())))
```

Comment:
This is probably the culprit of the bad performance. This should be:
```suggestion
        .ok_or_else(|| HintError::UnknownIdentifier(Box::new(var_name.to_string())))
```